modules/c++/nitf/source/ByteProvider.cpp

```diff
@@ -432,12 +431,10 @@ void ByteProvider::addImageData(
                 rowsInSegmentSkipped * mNumBytesPerRow;
     }
 
-    const size_t numPadRows = countBytesForImageData(seg, numRowsToWrite,
-            imageDataEndRow) / mNumBytesPerPixel;
-    numRowsToWrite += numPadRows;
-    numPadRowsSoFar += numPadRows;
+    const size_t bytesThisWrite = countBytesForImageData(seg, numRowsToWrite,
```

Comment:
I think `numPadRows` is a leftover from my botched refactoring. (See line 393). It gives the correct byte count, so we should just use it instead of whatever this what attempting.